src/auto_archiver/archivers/base_archiver/base_archiver.py

```diff
@@ -0,0 +1,291 @@
+            result.set("upload_date", upload_date)
+        
+        # then clean away any keys we don't want
+        for clean_key in self.keys_to_clean(extractor_key, video_data):
+            video_data.pop(clean_key, None)
+        
+        # then add the rest of the video data
+        for k, v in video_data.items():
+            if v:
+                result.set(k, v)
+
+        return result
+
+    def video_data_metadata_mapping(self, extractor_key: str, video_data: dict) -> dict:
+        """
+        Returns a key->value mapping to map from the yt-dlp produced 'video_data' to the Metadata object.
+        Can be either a string for direct mapping, or a function, or a lambda.
+        """
+        return {}
+    
+    def suitable_extractors(self, url: str) -> list[str]:
+        """
+        Returns a list of valid extractors for the given URL"""
+        for info_extractor in yt_dlp.YoutubeDL()._ies.values():
+            if info_extractor.suitable(url) and info_extractor.working():
+                yield info_extractor
+        
+    def suitable(self, url: str) -> bool:
+        """
+        Checks for valid URLs out of all ytdlp extractors.
+        Returns False for the GenericIE, which as labelled by yt-dlp: 'Generic downloader that works on some sites'
+        """
+        return any(self.suitable_extractors(url))
+
+    def create_metadata_for_post(self, info_extractor: InfoExtractor, video_data: dict, url: str) -> Metadata:
+        """
+        Standardizes the output of the ytdlp InfoExtractor to a common format
+        """
+        if info_extractor.ie_key() == 'Bluesky':
+            return bluesky.create_metadata(video_data, self, url)
+        if info_extractor.ie_key() == 'Twitter':
+            return twitter.create_metadata(video_data, self, url)
+
+    def get_metatdata_for_post(self, info_extractor: Type[InfoExtractor], url: str, ydl: yt_dlp.YoutubeDL) -> Metadata:
+        """
+        Calls into the ytdlp InfoExtract subclass to use the prive _extract_post method to get the post metadata.
+        """
+
+        ie_instance = info_extractor(downloader=ydl)
+        post_data = None
+
+        if info_extractor.ie_key() == 'Bluesky':
+            # bluesky kwargs are handle, video_id
+            handle, video_id = ie_instance._match_valid_url(url).group('handle', 'id')
+            post_data = ie_instance._extract_post(handle=handle, post_id=video_id)
+        elif info_extractor.ie_key() == 'Twitter':
+            # twitter kwargs are tweet_id
+            twid = ie_instance._match_valid_url(url).group('id')
+            # TODO: if ytdlp PR https://github.com/yt-dlp/yt-dlp/pull/12098 is merged, change to _extract_post
+            post_data = ie_instance._extract_status(twid=twid)
+
+        elif info_extractor.ie_key() == 'TikTok':
+            pass
+
+        else:
+            # lame attempt at trying to get data for an unknown extractor
```

Comment:
```suggestion
            # lame attempt at trying to get data for an unknown extractor
            # TODO: test some more video platforms and see if there's any improvement to be made
```
in case this has not been tested for others.